aries_vcx/src/protocols/connection/invitee/state_machine.rs

```diff
@@ -340,10 +283,48 @@ impl SmConnectionInvitee {
         })
     }
 
-    pub fn handle_connection_response(self, response: SignedResponse) -> VcxResult<Self> {
+    pub async fn handle_connection_response(
+        self,
+        response: SignedResponse,
+        send_message: SendClosureConnection
+    ) -> VcxResult<Self> {
         verify_thread_id(&self.get_thread_id(), &A2AMessage::ConnectionResponse(response.clone()))?;
         let state = match self.state {
-            InviteeFullState::Requested(state) => InviteeFullState::Responded((state, response).into()),
+            InviteeFullState::Requested(state) => {
+                let remote_vk: String = state.did_doc.recipient_keys().get(0).cloned().ok_or(VcxError::from_msg(
+                    VcxErrorKind::InvalidState,
+                    "Cannot handle response: remote verkey not found",
+                ))?;
+
+                match decode_signed_connection_response(response.clone(), &remote_vk).await {
```

Comment:
👍  Nice, finally having this at the right spot :-) Please include this change also in PR description